jdbc-driver/src/systemTest/java/com/google/cloud/dataproc/jdbc/DataprocSystemTest.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.dataproc.v1beta2.ClusterControllerClient;
+import com.google.cloud.dataproc.v1beta2.ClusterControllerSettings;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * This class tests end-to-end usage of the DataprocDriver. It tests the workflow for cluster
+ * creation with init action and Component Gateway enabled, connection establishment using various
+ * JDBC URL string, query execution with Hive, and cluster deletion after testing.
+ */
+public class DataprocSystemTest {
+    private static final String REGION = System.getProperty("region");
+    private static final String PROJECT_ID = System.getProperty("projectId");
+    private static final String USER = System.getProperty("user");
+    private static ClusterControllerClient clusterControllerClient;
+    private static Map<String, Map<String, String>> labelsByClusterName;
+    private static String cluster1;
+    private static String cluster2;
+    private static String cluster3;
+    private static String cluster4;
+
+    @BeforeClass
+    public static void setUp() throws IOException, ExecutionException, InterruptedException {
+        cluster1 = String.format("%s-test-cluster1", USER);
+        cluster2 = String.format("%s-test-cluster2", USER);
+        cluster3 = String.format("%s-test-cluster3", USER);
+        cluster4 = String.format("%s-test-cluster4", USER);
+
+        String myEndpoint = String.format("%s-dataproc.googleapis.com:443", REGION);
+        ClusterControllerSettings clusterControllerSettings =
+                ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+        clusterControllerClient = ClusterControllerClient.create(clusterControllerSettings);
+
+        deleteClusters();
```

Comment:
Nit: comment about why this is necessary and how it only cleans up clusters from the same user (prefix)